functionary/core/models/workflow_step.py

```diff
@@ -0,0 +1,95 @@
+import json
+import uuid
+from typing import TYPE_CHECKING
+
+from django.core.validators import RegexValidator
+from django.db import models, transaction
+from django.template import Context, Template
+
+from core.models import Task, Workflow, WorkflowRunStep
+
+if TYPE_CHECKING:
+    from core.models import WorkflowRun
+
+
+VALID_STEP_NAME = RegexValidator(
+    regex="^[a-zA-Z0-9_]+$",
```

Comment:
I think this could just be simplified to `^\w+$`

Reply:
Agreed, this wasn't used in the Variables since we wanted those to be all uppercase.